미션 생성, 조회, 인증

devyubin opened the pull request:
## ✅ 풀\_리퀘스트 체크리스트

<!--
하나씩 확인 후 체크박스에 표시해주세요.
-->

- [x] PR 제목: [Feature/Fix/Refactor...] 작업 내용 한 줄 요약 (브랜치 이름) #이슈번호
- [x] commit message 가 적절한지 확인해주세요.
- [x] 적절한 branch 로 요청했는지 확인해주세요.
- [x] Assignees, Label 을 붙여주세요.
- [x] 주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 Reviewer 로 등록해주세요.
- [x] PR이 승인된 경우 해당 브랜치는 삭제해 주세요!

<br/>

## 🔄 변경 사항
미션 생성 및 조회를 구현 및 테스트하였습니다.
- 미션 생성
- 미션 조회 (내가 받은 / 내가 준 / 수행중인 / 끝난)
- 미션 인증

<br/>

<!-- 관련되어있는 Issue Number 를 작성하세요! 해당 이슈를 이곳에 적으면 pr merge 이후 해당 이슈는 자동으로 close 됩니다. -->

closes: #12 

## 👨🏻‍💻 테스트 체크리스트
- 미션 등록이 성공적으로 수행된다.
- 진행 중인 미션을 성공적으로 조회한다.
- 완료된 미션을 성공적으로 조회한다.
- 미션 주차가 현재 주차보다 이전일 경우 예외가 발생한다.

<br/>

## 📌 변경 및 주의 사항
- 따로 없습니다.
<!--
변경사항 및 주의 사항이 있다면 적어주세요.
주의 사항과 관련해 꼭 확인해야 할 사람이 있다면 리뷰어로 등록해주세요. (다른 사람이 작성한 코드 수정 등)
코드 리뷰 시 더 꼼꼼하게 확인 받고 싶은 부분이 있다면 적어주세요.
-->

<br/>

devyubin committed:
Merge remote-tracking branch 'origin/main' into feature/mission

devyubin committed:
Fix: 미션 정상 저장을 위한 엔티티 수정

현재 엔티티에서 미션 등록시 giver/receiver 관련 문제 해결을 위해 엔티티 일부를 수정했습니다.

devyubin committed:
Feat: 미션 생성 및 조회 (#12)

아래 기능의 구현 및 테스트를 작성하였습니다.
- 미션 생성
- 미션 조회

devyubin commented:
웹훅 테스트입니다.

devyubin commented:
test

Hawardshin commented on src/main/java/com/techwave/olol/mission/domain/Mission.java:
```diff
@@ -24,17 +26,22 @@
 @SQLDelete(sql = "UPDATE Mission SET is_delete = true WHERE id = ?")
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
+@AllArgsConstructor
+@Builder
+@Getter
```
이거 Buider를 최상위에 선언하면 successStamps에서 초기화 표현식이 막혀어 테스트 할 때 NPE 날 거에요!
아래 있는 주석 지워주시면서 successStamps아래에 
@Builder.Default 로 해주시면 테스트가 안 깨질 것 같아요!
지금 SuccessStamp RepositoryTest에서 미션에 새로운 스탬프를 저장할 수 있다에 오류가 나요!

devyubin replied:
확인했습니다. 수정해서 커밋할게요!

Hawardshin reviewed (approved):
고생 많으셨습니다~! 
지금 CheckStyle에 몇개가 import-grouping으로 잡히는데 확인 한번만 부탁드릴께요~!

devyubin committed:
Fix: successStamps 오류 해결

아래 테스트에서 발생하는 초기화 표현식 오류를 해결하였습니다.
- SuccessStamp RepositoryTest에서 미션에 새로운 스탬프를 저장할 수 있다

devyubin committed:
Feat: 미션 인증 (#12)

아래 기능의 구현 및 테스트를 작성하였습니다.
- 미션 인증

devyubin committed:
Fix: 미션 인증 로직 수정

- 미션 인증 관련 내용이 SuccessStamp 에 저장되지 않는 문제점 수정
- successQuora 증가 메서드 삭제

devyubin committed:
Test: 미션 인증 테스트 추가

Hawardshin commented on src/main/java/com/techwave/olol/mission/controller/MissionController.java:
```diff
@@ -34,28 +35,39 @@ public class MissionController {
+		@PathVariable("missionId") UUID missionId,
+		@RequestParam("file") MultipartFile file) {
+		try {
+			missionService.verifyMission("dummyNickname", missionId, file);
```
이거 지금 security 안되셔 임시로 이렇게 해둔건가요?

devyubin replied:
넵 맞습니다! security 이슈 확인 되는 이후 바로 수정할 예정이에요.

Hawardshin reviewed (approved):
고생하셨습니다~~
